app/app/search/SearchMenu.tsx

```diff
@@ -21,6 +21,12 @@ const SearchMenu = () => {
     });
   };
 
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter") {
+      handleSearch(e as unknown as React.MouseEvent);
```

Comment:
don't cast it, it won't work. just call `e.preventDefault()` here and split the `handleSearch` into two separate functions — `handleSearch()` and `handleClick(e: React.MouseEvent)`

Reply:
Don't let search happen when mutating — you have to add checks similar to the button disabled (but put them in the handler instead of here